index.js

```diff
@@ -141,7 +141,11 @@ function renderTag(elem, opts) {
   ) {
     tag += '/>';
   } else {
-    tag += '>';
+    if (singleTag[elem.name] && opts.recognizeSelfClosing) {
```

Comment:
No, I simply dislike having two identical branches. I would also opt for reusing the children check from the previous branch, as that also includes tags wich simply lack any content.

(The change would then consist of replacing `opts.xmlMode` with `(opts.xmlMode || opts.recognizeSelfClosing)`.)